src/main/java/com/zendesk/maxwell/producer/NatsProducer.java

```diff
@@ -0,0 +1,68 @@
+package com.zendesk.maxwell.producer;
+
+import com.zendesk.maxwell.MaxwellContext;
+import com.zendesk.maxwell.row.RowMap;
+import com.zendesk.maxwell.util.TopicInterpolator;
+import io.nats.client.Connection;
+import io.nats.client.Nats;
+import io.nats.client.Options;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.List;
+
+public class NatsProducer extends AbstractProducer {
+
+	private final Logger LOGGER = LoggerFactory.getLogger(NatsProducer.class);
+	private final Connection natsConnection;
+	private final String natsSubjectTemplate;
+
+	public NatsProducer(MaxwellContext context) {
+		super(context);
+		try {
+			List<String> urls = Arrays.asList(context.getConfig().natsUrl.split(","));
+
+			Options.Builder optionBuilder = new Options.Builder();
+
+			urls.forEach(optionBuilder::server);
+
+			Options option = optionBuilder.build();
+
+			this.natsConnection = Nats.connect(option);
+			this.natsSubjectTemplate = context.getConfig().natsSubject;
+
+		} catch (IOException | InterruptedException e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	@Override
+	public void push(RowMap r) throws Exception {
+		if (!r.shouldOutput(outputConfig)) {
+			context.setPosition(r.getNextPosition());
+			return;
+		}
+
+		String value = r.toJSON(outputConfig);
+		String natsSubject = new TopicInterpolator(this.natsSubjectTemplate).generateFromRowMapAndTrimAllWhitesSpaces(r);
+
+		long maxPayloadSize = natsConnection.getMaxPayload();
```

Comment:
have you looked at the source for this method?  Are you sure that `getMaxPayload()` doesn't make a network request?

Reply:
From what I saw in the source code of [getMaxPayload](https://github.com/nats-io/nats.java/blob/master/src/main/java/io/nats/client/impl/NatsConnection.java#L1503), it just return the value of a field from `Connection.serverInfo`, which was already achieved when we created the connection, so no network request for me :)